sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/StreamUtilsImpl.java

```diff
@@ -28,6 +30,20 @@ public byte[] readAllBytes(InputStream inputStream) throws IOException {
         return outputStream.toByteArray();
     }
 
+    @Override
+    public long transfer(InputStream in, OutputStream out) throws IOException {
```

Comment:
Mind writing a comment to potentially optimize this if this will be called often. For example `ByteArrayInputStream` will have a known length and durable backing byte array which should be transferable in a single call.